src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java

```diff
@@ -106,6 +110,49 @@ public Try<UserId> validateUserToken(String carbonioUserToken) {
     }
   }
 
+  /**
+   * Allows to retrieve a list of Carbonio users by {@link UUID} if they exist.
+   *
+   * @param cookie    is a {@link String} representing the requester cookie. (It is necessary to
+   *                  have only the ZM_AUTH_TOKEN cookie).
+   * @param usersUUID is a {@link List} of {@link UUID}s of the Carbonio users to retrieve.
+   * @return a {@link Try#success} containing a {@link List} of {@link UserInfo} representing
+   * the related Carbonio users if they exist. <p />It returns a {@link Try#failure} containing
+   * an {@link UserNotFound} throwable the cookie is not valid. <p />It returns a
+   * {@link Try#failure} containing an {@link InternalServerError} throwable if something goes
+   * wrong during the API call.
+   */
+  public Try<List<UserInfo>> getUsers(
+    String cookie,
+    List<UUID> usersUUID
+  ) {
+    CloseableHttpClient httpClient = HttpClients.createMinimal();
+
+    AtomicReference<String> getRequest = new AtomicReference<>(userManagementURL + getUsersEndpoint + "?userIds=");
+    usersUUID.forEach(userUUID -> getRequest.updateAndGet(v -> v + userUUID));
+    HttpGet request = new HttpGet(getRequest.get());
+    request.setHeader("Cookie", cookie);
+
+    try {
+      CloseableHttpResponse response = httpClient.execute(request);
+
+      if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+        String bodyResponse = IOUtils.toString(
+          response.getEntity().getContent(),
+          StandardCharsets.UTF_8
+        );
+
+        return Try.success(new ObjectMapper().readValue(bodyResponse, new TypeReference<>() {}));
+      }
+
+      return Try.failure(
+        new UserNotFound("The cookie: \"" + cookie + "\" is not valid")
```

Comment:
I think this is not the correct error you should return here, it depends on how you handle this request on the user-management side. Infact I noticed that the yaml on the carbonio-user-management side has only 400 bad request error, so here you couldn't use "The cookie is not valid" as message error.